service/src/main/java/bio/terra/profile/service/crl/GcpCrlService.java

```diff
@@ -33,4 +37,14 @@ public CloudBillingClientCow getBillingClientCow(AuthenticatedUserRequest user)
   private GoogleCredentials getUserCredentials(String token) {
     return GoogleCredentials.newBuilder().setAccessToken(new AccessToken(token, null)).build();
   }
+
+  private String getGoogleAccessToken(String userToken) {
+    try {
+      var jwt = SignedJWT.parse(userToken);
+      return (String) jwt.getJWTClaimsSet().getClaim(GOOGLE_ACCESS_TOKEN_CLAIM);
```

Comment:
I generally stay away from parsing the JWT since infosec cares very much that you do it right. This may be checking the signature, but it isn't checking the issuer or audience or other things they may care about.

Did you know that the apache proxy extracts all claims in the B2C JWT and adds them to request headers prefixed with `OAUTH2_CLAIM_`. You can find what you need in the `OAUTH2_CLAIM_idp_access_token` request header instead of parsing the JWT yourself.

Reply:
BPM doesn't run with a proxy when it runs locally which was the motivation for doing it this way. Maybe it would be better to only get the idp access token in this fashion if BPM is running locally and rely on the request header in all other cases.

My hope was that the user's token has been properly vetted by the time we get to this point and there was no harm in trying to grab this claim if it existed, but I will see what infosec has to say about that.

Reply:
disregard this comment